rfd/0029-account-lifecycle.md

```diff
@@ -0,0 +1,220 @@
+and if lost, the account can not be recovered.
+
+### Recovery scenarios
+
+The first two recovery scenarios work for every system user, reducing the load
+on Teleport system administrators as well.
+
+**Emails and usernames**
+
+Teleport Cloud assumes that username is a user email.
+When sending emails, Teleport cloud will send an email to the username if it contains @ sign.
+Self hosted Teleport will not send any emails.
+
+**Account owners**
+
+Teleport considers any user with write `billing` resource permission
+as account owner. This doc refers to those users as account owner users.
+
+We should recommend to have several account owners, who do not have system administration
+privileges - can't modify roles, or users.
+
+Any other users are referred to as simply users.
+
+**User lost a second factor**
+
+If any user lost a second factor, but not password, they can add a new second factor using
+a recovery token:
+
+* Login with a username and password.
+* Instead of pressing second factor, enter a one-time recovery token
+* Enroll new second factor.
+* All recovery tokens are no longer valid. The new set of recovery tokens are printed to the user.
+* User has to confirm that they have copied recovery tokens.
+* The session is not initiated. A user is redirected to a login form
+with UI displaying a message to login using new credentials.
+
+* If the operation is aborted after user enters the token, but fails to enter the password,
+the valid token is no longer used, but remaining tokens can be used.
+* If a user tries to reuse the token, 
+
+This flow ensures two factors are verified: the valid password and a recovery token.
+
+**User lost a password**
+
+If user lost a password, but not a second factor, they could recover using
+a recovery link sent to their email and a second factor:
+
+* User enters their username and clicks forgot password.
+* User clicks have to present a valid second factor and a recovery token,
+after that they will be able to reset their password.
+* All recovery tokens are no longer valid. The new set of recovery tokens are printed to the user.
+* User has to confirm that they have copied recovery tokens.
+* The session is not initiated. A user is redirected to a login form
+with UI displaying a message to login using new credentials.
+
+This flow ensure that two factors are verified:
+
+* Second factor TOTP token or U2F device.
+* A valid account recovery token.
+
+* We do not consider email as a valid verification factor. It's a nice-to-have additional protection layer.
+The email transport is not guaranteed to be secure, and email itself can be hacked.
+
+**Getting new tokens**
+
+If any user is logged into Teleport UI, they can regenerate new password recovery tokens:
+
+* Alice goes to her account settings and chooses to generate new set of recovery tokens.
+* Teleport asks for a second factor before generating tokens.
+* A new set of tokens is printed for Alice. The message makes it clear
```

Comment:
The message should probably come before or both before and when shown the tokens.